automate/tests/test_views.py

```diff
@@ -89,10 +94,58 @@ def test_to_retrieve_and_update_project(self):
 
         with self.subTest("Update project"):
             response = self.client.patch(
-                self.url_detail,
-                data={"primary_repo_type": RepoTypeChoices.BITBUCKET.value},
+                self.url_detail, data={"primary_repo_type": RepoTypeChoices.BITBUCKET.value}
             )
             content = response.data
-            self.assertEqual(
-                RepoTypeChoices.BITBUCKET.value, content.get("primary_repo_type")
-            )
+            self.assertEqual(RepoTypeChoices.BITBUCKET.value, content.get("primary_repo_type"))
+
+
+class TestWebhook(APITestCase):
+    """Test for Webhook."""
+
+    def setUp(self) -> None:
+        self.user = UserFactory()
+        self.project = ProjectFactory(owner=self.user)
+        self.url = reverse(
+            "repository:project-webhook", kwargs={"slug": self.project.slug}
+        )
+        self.data = {
+            "action": "closed",
+            "pull_request": {
+                "id": fake.random_number(9),
+                "url": fake.url(),
+                "state": "closed",
+                "title": fake.text(30),
+                "body": fake.sentence(30),
+                "head": {"ref": fake.text(15), "repo": {"name": fake.text(10)}},
+            },
+        }
+
+    def test_to_ensure_webhook_gets_data(self):
+        # this test throws a validation if webhook is called without data
+        self.client.force_authenticate(self.user)
+        response = self.client.post(self.url)
+        content = response.json()
+
+        self.assertTrue(response.status_code == 400)
+        self.assertEqual(content.get("action"), ["This field is required."])
+        self.assertEqual(content.get("pull_request"), ["This field is required."])
+
+    @patch("automate.gitremote.GitRemote.run")
+    def test_to_ensure_webhook_works(self, run):
+        self.client.force_authenticate(self.user)
+
+        with patch("automate.gitremote.GitRemote.run") as mock_run:
```

Comment:
U dont need to mock the same thing twice, You can use `run` instead of this new context_manager

Reply:
Thanks. done